apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PgNumeric.cs

```diff
@@ -0,0 +1,487 @@
+            int scale = 0;
+            // If we have '.', then we need to extract the value and scale.
+            if (pointIndex != -1)
+            {                
+                int decimalPlaces = text.Length - pointIndex - 1;
+                text = text.Remove(pointIndex, 1);
+                scale = decimalPlaces;
+                rawValue = BigInteger.Parse(text, CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                // We don't have '.'
+                rawValue = BigInteger.Parse(text, CultureInfo.InvariantCulture);
+            }
+            if (!IsRawValueInRange(rawValue, scale))
+            {
+                value = default;
+                return "Parsed value would overflow the range of the type.";
+            }
+            value = new PgNumeric(rawValue, scale, false);
+            return null;
+        }
+
+        /// <summary>
+        /// Removes trailing zeros from the text.
+        /// Should be called only for text with '.'.
+        /// 1.0000 will be returned as 1, 1.00100 will be returned as 1.001.
+        /// </summary>
+        /// <param name="text">The text from which trailing zeros need to be trimmed.</param>
+        /// <returns>The text without trailing zeros.</returns>
+        private static string RemoveTrailingZeros(string text)
+        {
+            if (text.IndexOf('.') != -1)
+            {
+                StringBuilder stringBuilder = new StringBuilder(text);
+                while (stringBuilder.Length > 0 && stringBuilder[stringBuilder.Length - 1] == '0')
+                {
+                    stringBuilder.Length--;
+                }
+
+                // Handle the case that after trimming 0, number is ending with '.'
+                if (stringBuilder.Length > 0 && stringBuilder[stringBuilder.Length - 1] == '.')
+                {
+                    stringBuilder.Length--;
+                }
+
+                return stringBuilder.ToString();
+            }
+
+            // We don't have '.', return original text.
+            return text;
+        }
+
+        /// <summary>
+        /// Returns a canonical string representation of this value. This always uses "." as a decimal point.
+        /// If the value is between -1 and 1 exclusive, a "0" character is included before the decimal point.
+        /// </summary>
+        /// <returns>A canonical string representation of this value.</returns>
+        public override string ToString()
+        {
+            if (_notANumber)
+            {
+                return NotANumber;
+            }
+
+            int sign = _scaledValue.Sign;
+            if (sign == 0)
+            {
+                return "0";
+            }
+
+            var integerText = _scaledValue.ToString(CultureInfo.InvariantCulture);
+            int signLength = sign > 0 ? 0 : 1;
+
+            // We can have a scaled value or a non-scaled value, check if its scaled. 
+            if (_scale != 0)
```

Comment:
I'd suggest changing this to:

```csharp
if (_scale == 0)
{
    return integerText;
}
```

... then the rest of the code doesn't need to be indented, and it makes it clearer that that's basically an optimization path.